lte/gateway/docker/s1ap/readme.md

```diff
@@ -0,0 +1,156 @@
+
+* Ubuntu 20.04 x86
+* t2.large
+* Security group
+  * Allow SSH from My IP, or private address of your bastion host. **Important to not allow Public SSH access**
+  * Allow all traffic from 192.168.60.0/24
+  * Allow all traffic from 192.168.128.0/23
+* 3 Interfaces
+  * eth0: Auto-assign (subnet-management)
+    * Add Public IP
+  * eth1: 192.168.60.141 (subnet-sgi)
+  * eth2: 192.168.128.11 (subnet-s1)
+* 50 GB Disk
+* Key Pair (kp1)
+
+### Traffic Generator instance
+
+* Ubuntu 20.04 x86
+* t2.large
+* Security group
+  * Allow SSH from My IP, or private address of your bastion host. **Important to not allow Public SSH access**
+  * Allow all traffic from 192.168.60.0/24
+  * Allow all traffic from 192.168.128.0/23
+* 3 Interfaces
+  * eth0: Auto-assign (subnet-management)
+    * Add Public IP
+  * eth1: 192.168.60.144 (subnet-sgi)
+  * eth2: 192.168.129.42 (subnet-s1)
+* 50 GB Disk
+* Key Pair (kp1)
+
+## Add your ssh key pair (kp1) used when creating the instances to your ssh-agent
+
+```
+ssh-add .ssh/id_rsa
+ssh-add -l
+3072 SHA256:KWaiSLPiTYy5tRGXSKpkuwJcYDGy92NTMi0v3khLjJY (RSA)
+```
+
+## Run S1APTester playbook
+
+### ssh to the agw instance and deploy as a containerized AGW
+
+```
+# ssh to AGW instance
+ssh -A ubuntu@184.72.47.28
+
+# Download containerized AGW bootstrap script
+wget https://raw.githubusercontent.com/magma/magma/master/lte/gateway/deploy/agw_install_docker.sh
+
+# Create certs directory and populate with your orc8r rootCA
+sudo mkdir -p /var/opt/magma/certs
+sudo su root -c 'sudo echo "-----BEGIN CERTIFICATE-----
+-----END CERTIFICATE-----" > /var/opt/magma/certs/rootCA.pem'
+
+# Run bootstrap script and reboot
+sudo su -c 'bash -x agw_install_docker.sh'
+sudo reboot
+```
+
+### After reboot, add the s1ap and trfgen ssh keys to known_hosts of the AGW Instance
+
+```
+ssh -A ubuntu@184.72.47.28
+ssh-keyscan -H 192.168.60.144 >> ~/.ssh/known_hosts
+ssh-keyscan -H 192.168.60.141 >> ~/.ssh/known_hosts
+```
+
+### move to deploy folder and test your connection to s1ap and trfgen instances
+
+```
+cd /opt/magma/lte/gateway/deploy
+
+ansible -m ping all -i agw_hosts -b
+127.0.0.1 | SUCCESS => {
+192.168.60.141 | SUCCESS => {
+192.168.60.144 | SUCCESS => {
+```
+
+### Deploy s1aptester
+
+Depending on whether you already have your AGW and s1aptest images in your registry, or want to build them, your can supply the `docker_registry` environment variable to the playbook. If you don't supply the variable, it will build the images and start the services. If you supply it, it will pull down the images from the registry and start the services.
+
+#### Without registry
+
+```
+ansible-playbook -i agw_hosts magma_docker_s1ap_setup.yml -b
+```
+
+#### With registry
+
+```
+ansible-playbook -i agw_hosts magma_docker_s1ap_setup.yml -b -e "docker_registry=public.ecr.aws/yourrepo/"
+```
+
+### Running tests Commands
+
+#### ssh to the s1aptester instance and attach to the s1aptester container to run tests
+
+```
+ssh 192.168.60.141
```

Comment:
This looks like an "internal ssh" from one AWS instance to another.
It should be clarified, what the source and target of this ssh are.